tests/components/voice_assistant/test_pipeline.py

```diff
@@ -0,0 +1,108 @@
+"""Pipeline tests for Voice Assistant integration."""
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from homeassistant.components import tts
+from homeassistant.components.voice_assistant.pipeline import (
+    AudioPipelineRequest,
+    Pipeline,
+    PipelineEventType,
+    PipelineRun,
+)
+from homeassistant.core import Context
+from homeassistant.setup import async_setup_component
+
+
+@pytest.fixture(autouse=True)
+async def init_components(hass):
+    """Initialize relevant components with empty configs."""
+    assert await async_setup_component(hass, "voice_assistant", {})
+
+
+@pytest.fixture
+async def mock_get_tts_audio(hass):
+    """Set up media source."""
+    assert await async_setup_component(hass, "media_source", {})
+    assert await async_setup_component(
+        hass,
+        "tts",
+        {
+            "tts": {
+                "platform": "demo",
+            }
+        },
+    )
+
+    with patch(
+        "homeassistant.components.demo.tts.DemoProvider.get_tts_audio",
+        return_value=("mp3", b""),
+    ) as mock_get_tts:
+        yield mock_get_tts
+
+
+async def test_audio_pipeline(hass, mock_get_tts_audio):
+    """Run audio pipeline with mock TTS."""
+    pipeline = Pipeline(
+        name="test",
+        language=hass.config.language,
+        conversation_engine=None,
+        tts_engine=None,
+    )
+
+    event_callback = MagicMock()
+    await AudioPipelineRequest(intent_input="Are the lights on?").execute(
+        PipelineRun(
+            hass,
+            context=Context(),
+            pipeline=pipeline,
+            event_callback=event_callback,
+            language=hass.config.language,
+        )
+    )
+
+    # Clean up demo mp3
+    await hass.services.async_call(
```

Comment:
Instead of this, we should never write the MP3 to begin with. 

Import the fixture `mock_init_cache_dir` from `homeassistant.components.tts.conftest`